### PR TITLE
User registration web handler

### LINKED ISSUE #152: User registration API
## Task Summary
- User registration API using which user can be registered.
- API will accept all necessary json data for registration through HTTP request and return appropriate json response.

## Acceptance Criteria
- [x] API must accept data for registration
- [x] Registration with given data only.
- [x] Duplicate registration should not happen
- [x] Handle username (Auto generate/ Accept username), but it must be unique
- [x] Send response with success or failure status. 

## Subtasks
- NA.
  
## Related Issues
- NA

## Additional Notes
- Basic and necessary validation must be handled, _such as unique `username`_

### PR DESCRIPTION
**Pull request by priyapal2000**
## Description
The user registration web handler processes incoming registration requests, store user data in the database and send the message registration successfully.
## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Project Configuration
- [x] Repository Configuration
- [ ] Documentation update - Code guide/ docs (_like, javadoc or api_)
- [ ] Other (please describe):

## Screenshots / GIFs / Media (if applicable)
- NA

## How Has This Been Tested?/ Test Plan
   Use Postman, tool to send a POST request to /register.
   Verify:
    1.Correct status codes and responses for valid and invalid inputs.
    2.The user is successfully saved in the database.

## Checklist:
- [x] My code follows the project’s [coding guidelines](./../../docs/guidelines/coding-guides/). 
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code where necessary.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] I have added tests that verify the functionality of the code (if applicable).
- [ ] All new and existing tests pass.

## How to use/ enable
  Endpoint: POST /api/register
  Request Body:
  ```json
{
    "address": "Pune",
    "city": "Malad",
    "dob": "2000-02-06",
    "email": "priya@gmail.com",
    "moNo": 36547892,
    "name" : "riya",
    "pincode": 400097,
    "enabled": 1,
    "username": "priyapal"
}
```
JSON output success:
User saved successfully

JSON output alreaday exit:
User already exists 

JSON output failure:
User not save successfully. Please check your input and try again.
> **TODO: Add response body (JSON ouput) for failure and success**

## Additional Notes
- NA

## Closing issues

closes #152

### CODE REVIEW

File: Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import com.arcitech.model.User;
+
+/**
+ * @author Priya
+ * 
+ */
+public interface UserService {
+	
+	public String addUser (User user);
+
```
Comment (ajstyle2410) [inline Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java:17]: remove unwanted space

**Comment by ajstyle2410** [comment]
update the service file  resolve the error   and remove the unwanted spaces in the code inside use  streams inside like as Optional  streams

**Comment by ajstyle2410** [comment]
check whether the username and email exist  or not  in the database inside

**Comment by ajstyle2410** [comment]
add the task description

File: Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java (hunk 1 above)
Comment (priyapal2000) [reply Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java:17]: remove unwanted space is done.

**Comment by ajstyle2410** [comment]
1. resolve the Quality Gate failed  
2. resolve the pull request  template questions 

 resolve the Quality Gate failed is done know is Gate passed.

File: Arc-i-Tech/src/main/java/com/arcitech/helper/RandomPasswordGenerater.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+			+ "abcdefghijklmnopqrstuvxyz"; 
+			// create StringBuffer size of AlphaNumericString 
+		StringBuilder sb = new StringBuilder(n); 
+
+		for (int i = 0; i < n; i++) { 
+			// generate a random number between 
+			// 0 to AlphaNumericString variable length 
+			int index 
+			= (int)(AlphaNumericString.length() 
+			* Math.random());
```
Comment (ajstyle2410) [inline Arc-i-Tech/src/main/java/com/arcitech/helper/RandomPasswordGenerater.java]: replace  Random class to  SecureRandom  class

**Review by ajstyle2410 — CHANGES_REQUESTED** [review]
remove unwanted spaces

File: Arc-i-Tech/src/main/java/com/arcitech/helper/RandomPasswordGenerater.java (hunk 2 above)
Comment (priyapal2000) [reply Arc-i-Tech/src/main/java/com/arcitech/helper/RandomPasswordGenerater.java]: In replace Random class is changes is done.

**Comment by ajstyle2410** [comment]
@priyapal2000  add the new pull request template

**Comment by priyapal2000** [comment]
@ajstyle2410 
Added new pull request template is Done.

**Comment by ajstyle2410** [comment]
@priyapal2000 
filled in the pull request  details

**Comment by priyapal2000** [comment]
@ajstyle2410 
PR updated is done.

**Comment by priyapal2000** [comment]
@ajayg2808 
PR update is done

**Comment by priyapal2000** [comment]
@ajayg2808 
PR update is done.

**Comment by ajayg2808** [comment]
@priyapal2000 
- [x] Complete `TODO` part in PR description
- [x] Remove PR from GitHub project (PR is not tracked in project only issue is tracked)
- [ ] Remove/ revert unnecessary files/ commit. (I'll start code review once you do this change)

**Pull request closed without merging**